lighthouse-core/audits/seo/robots-txt.js

```diff
@@ -108,44 +108,55 @@ function parseLine(line) {
   };
 }
 
+/**
+ * @param {string} content
+ * @returns {Array<{index: string, line: string, message: string}>}
+ */
 function validateRobots(content) {
+  /**
+   * @type Array<{index: string, line: string, message: string}>
+   */
+  const errors = [];
   let inGroup = false;
 
-  return content
+  content
     .split(/\r\n|\r|\n/)
-    .map((line, index) => {
```

Comment:
I couldn't figure out how to explain to tsc that:

```
arr.map(() => {
  if(x) { return {} }
  return null;
}
.filter(x => x !== null)
```

can't possibly return `null`s because they are filtered out, so I rewrote it using `forEach`.